src/drive/web/modules/navigation/duck/actions.jsx

```diff
@@ -431,18 +436,51 @@ export const downloadFiles = files => {
   }
 }
 
+const encryptedDataToBlobURL = async file => {
+  // prepare decryption
+  const encryption = file.metadata
+  const iv = decodeArrayBuffer(encryption.iv)
+  const key = await importKeyJwk(encryption.key)
+  // Now fetch data
+  const data = await cozy.client.files
+    .downloadById(file.id || file._id)
+    .then(r => {
+      return r.blob()
+    })
+    .then(encryptedBlob => {
+      return new Response(encryptedBlob).arrayBuffer()
+    })
+    .then(encryptedBuffer => {
+      return decryptData(key, encryptedBuffer, { iv })
+    })
+  return URL.createObjectURL(new Blob([data], { type: file.type }))
+}
+
 const downloadFile = (file, meta) => {
-  return async dispatch => {
-    const downloadURL = await cozy.client.files
-      .getDownloadLinkById(file.id)
-      .catch(error => {
-        Alerter.error(downloadFileError(error))
-        throw error
-      })
-    const filename = file.name
+  const encrypted = file.metadata.iv !== undefined
```

Comment:
The iv is stored in `file.metadata.encryption` alongside the key: https://github.com/Gara64/cozy-drive/blob/crypto/src/drive/web/modules/upload/index.js#L211

Reply:
It was not but now it is in encryption-side and decryption-side

Reply:
Shame on me. Thanks!